netbox/netbox/settings.py

```diff
@@ -219,12 +224,24 @@
             raise e
 
         # Monkey-patch django-storages to fetch settings from STORAGE_CONFIG
-        def _setting(name, default=None):
-            if name in STORAGE_CONFIG:
-                return STORAGE_CONFIG[name]
-            return globals().get(name, default)
         storages.utils.setting = _setting
 
+    elif STORAGE_BACKEND == 'swift.storage.SwiftStorage':
+
+        try:
+            import swift.utils  # type: ignore
+        except ModuleNotFoundError as e:
+            if getattr(e, 'name') == 'swift':
+                raise ImproperlyConfigured(
+                    f"STORAGE_BACKEND is set to {STORAGE_BACKEND} but django-storage-swift is not present. "
+                    "It can be installed by running 'pip install django-storage-swift'."
+                )
+            raise e
+
+        # Monkey-patch django-storage-swift to fetch settings from STORAGE_CONFIG
+        swift.utils.setting = _setting
```

Comment:
Not a big fan of monkey-patching if it can be avoided. I'd rather handle this like [we do for django-social-auth](https://github.com/netbox-community/netbox/blob/develop/netbox/netbox/settings.py#L584-L587), by importing any global variables prefixed with `SWIFT_` from `configuration.py`.

Reply:
Indeed ! PR updated. I guess we could do the same with `django-storages` but I didn't want to risk breaking something external to my change as I don't use that lib.